kano_updater/apt_wrapper.py

```diff
@@ -259,13 +259,15 @@ def get_required_upgrade_space(self, priority=Priority.NONE):
 
         for pkg in self.upgradable_packages(priority=priority):
             state = AptPkgState.get_package_state(pkg)
-            orig_state.append((pkg, state))
 
-            pkg.mark_upgrade()
+            if state != AptPkgState.MARKED_UPGRADE:
+                orig_state.append((pkg, state))
+                pkg.mark_upgrade()
 
         space = self._cache.required_download + self._cache.required_space
 
-        for pkg, state in orig_state:
+        # Restore package states in reverse order
+        for pkg, state in reversed(orig_state):
             AptPkgState.restore_pkg_state(pkg, state)
```

Comment:
Do you want to put this `restore` under a similar `if` as above?

Reply:
Don't need to because they would have only been appended to `orig_state` if they weren't already marked for update.